pyuvdata/uvbeam/cst_beam.py

```diff
@@ -220,36 +226,127 @@ def read_cst_beam(
+            for name in data_col_enum:
+                this_col = np.where(np.array(column_names) == name)[0]
+                if this_col.size > 0:
+                    data_col = data_col + this_col.tolist()
+            if len(data_col) == 0:
+                raise ValueError(f"No power column found in file: {filename}")
+            elif len(data_col) > 1:
+                raise ValueError(
+                    f"Multiple possible power columns found in file: {filename}"
+                )
+            data_col = data_col[0]
+
+        tupled = list(zip(theta_data, phi_data))
+        n = len(tupled)
+        new_data = np.zeros((len(theta_axis) * len(phi_axis), data.shape[1]))
+        i = 0
+        idx = -1
+        for za in theta_axis:
+            for az in phi_axis:
+                if (idx + 1) < n and (za, az) == tupled[idx + 1]:
+                    # best case scenario is everything is in grid already.
+                    # For most beams, the large majority of iterations will hit this.
+                    idx = idx + 1
+                    new_data[i] = data[idx]
+                else:
+                    try:
+                        # Find the index of these coordinates
+                        # There may be multiple indices with the same coords, because
+                        # of poles -- here we just take the first one.
+                        idx = tupled.index((za, az))
+                        new_data[i] = data[idx]
+                        if np.isclose(new_data[i, theta_col], -np.pi):
+                            print(data[idx], za, az)
+                    except ValueError as err:
+                        # These co-ordinates do not exist.
+                        if za == 0 or za == np.pi:
+                            # Maybe we're at a pole, in which case we can potentially
+                            # deduce the data from other coordinates at the pole.
+                            if beam_type == "power":
+                                # We only care about abs() columns
+                                zero_idx = np.where(theta_data == 0)[0][0]
+                                new_data[i, data_col] = data[zero_idx, data_col]
+                            else:
+                                # We care about all the columns.
+                                idx = np.where(
+                                    np.isclose(theta_data, za)
+                                    & np.isclose(phi_data, (az + np.pi) % (2 * np.pi))
+                                )[0]
+                                if len(idx) == 0:
+                                    raise ValueError(
+                                        f"No coordinate data for za={za} with az={az}"
+                                    ) from err
+                                idx = idx[0]
+
+                                for icol, name in enumerate(column_names):
+                                    if "phase" in name:  # flip the phase.
```

Comment:
Can you expand the comment to explain why we're flipping the phase? Will help with understanding the code in the future.

Reply:
I actually don't really know the technical reasons, but I know it's true 😆 . I made the comment larger above here.

Reply:
I don't think you committed this change.